cmd/server/entrypoint.go

```diff
@@ -40,17 +48,45 @@ func StartProxySvr(cliCtx *cli.Context) error {
 		return fmt.Errorf("failed to pretty print config: %w", err)
 	}
 
+	var secretConfig common.SecretConfigV2
+	if cfg.EigenDAConfig.EdaClientConfigV2.Enabled {
+		// secret config is kept entirely separate from the other config values, which may be printed
+		secretConfig = eigendaflags_v2.ReadSecretConfigV2(cliCtx)
+		if err := secretConfig.Check(); err != nil {
+			return err
+		}
+	}
+
 	m := metrics.NewMetrics("default")
 
 	ctx, ctxCancel := context.WithCancel(cliCtx.Context)
 	defer ctxCancel()
 
-	sm, err := server.LoadStoreManager(ctx, cfg, log, m)
+	memConfig := cfg.EigenDAConfig.MemstoreConfig
+	if !cfg.EigenDAConfig.MemstoreEnabled {
+		memConfig = nil
+	}
+
+	sm, err := store.NewBuilder(
+		ctx,
+		cfg.EigenDAConfig.StorageConfig,
+		cfg.EigenDAConfig.EdaVerifierConfigV1,
+		cfg.EigenDAConfig.EdaClientConfigV1,
+		cfg.EigenDAConfig.EdaClientConfigV2,
+		secretConfig,
+		memConfig,
+		log,
+		m,
+	).BuildManager(
+		ctx,
+		cfg.EigenDAConfig.EdaClientConfigV2.PutRetries,
+		cfg.EigenDAConfig.MaxBlobSizeBytes,
+		cfg.EigenDAConfig.EigenDACertVerifierAddress)
```

Comment:
does the storeBuilder only have a single `BuildManager` function? maybe rename to just `Build`? Or the more explicit `BuildStoreManager` if there are other things. Otherwise `store.NewBuilder().BuildManager` feels like I'm not sure what exactly this is building.

Also why are we passing extra args to `BuildManager`? Shouldnt those be passed to the builder instead?

Reply:
Agreed, it doesn't make much sense like that.

I did some substantial renaming / reordering cleanup of the `Builder` in [bcd2379](https://github.com/Layr-Labs/eigenda-proxy/pull/321/commits/bcd237941f223348dab947e3aa6ca6820d300232), and also some minor changes in the code that uses the builder